### PR TITLE
Patcher Promotion Workflow docs

### PR DESCRIPTION
**Pull request by ZachGoldberg**

### CODE REVIEW

File: _docs-sources/patcher/getting-started/promotion-workflows.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+ls
+dev-account1  dev-account2  prod-account1  prod-account2  stage-account1  stage-account2
+```
+
+Then you would define your dev environments as `dev-*`, stage as `stage-*` and prod as `prod-*` and so fourth.
+
+
+# Implementation & Setup Example
+
+For the purposes of this we'll setup a promotion workflow that promotes through `dev`, `stage` and finally `prod`.
```
Comment (josh-padnick) [inline _docs-sources/patcher/getting-started/promotion-workflows.md:46]: s/setup/set up

File: _docs-sources/patcher/getting-started/promotion-workflows.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+## Setting up the initial dev promotion step
+
+The initial GitHub Actions Workflow file, which for this example lets call `update-dev.yml`, contains several key points
```
Comment (josh-padnick) [inline _docs-sources/patcher/getting-started/promotion-workflows.md:50]: Add `:`

File: _docs-sources/patcher/getting-started/promotion-workflows.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+* We setup the job to run on a schedule, to be a pull request target, a workflow dispatch and a repository dispatch.
+    * The schedule is of course optional but recommended
+    * The workflow dispatch is a recommended testing mechanism
```
Comment (josh-padnick) [inline _docs-sources/patcher/getting-started/promotion-workflows.md:54]: The term "workflow dispatch" is generic and confusing. Can we clarify this is triggered when a GitHub Action calls this workflow explicitly?

File: _docs-sources/patcher/getting-started/promotion-workflows.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    * The pull request target is required to trigger certain jobs
```
Comment (josh-padnick) [inline _docs-sources/patcher/getting-started/promotion-workflows.md:55]: This goes right to the technical details, but the high-level strategy isn't clear. Can you clarify the big picture here?

**Review by josh-padnick — DISMISSED** [review]
Looks great overall! Most of my feedback is around clarifying the big picture of what we're doing so that the reader has a clear understanding of what we're trying to accomplish, versus trying to infer that by understanding the technical details.

I'm excited to see this captured in docs!

File: _docs-sources/patcher/getting-started/promotion-workflows.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+
+# Patcher Promotion Workflow Architecture
+
+## Environments
+
+The concept of environments in Patcher Promotion Workflows is modelled via glob-pattern matched folder structures.  Patcher commands then accept commands to match those folders, such as `include_dirs: "{*prod*}/**"`.  A given environment can include 1 or many (without limit) folders.  Patcher will scan the entire group of folders at once for potential updates and changes.
+
+There is no limit on how many environments you can have, or other limit on the naming structure for those environments.
+
+In the future it is planned to model environments using a configuration based system (to be shared with Gruntwork Pipelines) which will allow for even more flexibility in your folder structure.
```
[the remark this answers is not in the document]
Comment (ZachGoldberg) [reply _docs-sources/patcher/getting-started/promotion-workflows.md:22]: When we actually update Patcher to consume those config files I suspect is when it'll be natural that these docs need updating.

File: _docs-sources/patcher/getting-started/promotion-workflows.md (hunk 3 above)
Comment (ZachGoldberg) [reply _docs-sources/patcher/getting-started/promotion-workflows.md:54]: "workflow dispatch" is a proper phrase in Github Actions-land and IMO is recognizable to those familiar with Actions.

**Pull request merged**